src/org/opensolaris/opengrok/web/Scripts.java

```diff
@@ -0,0 +1,238 @@
+            this.priority = priority;
+        }
+
+        abstract public String toHtml();
+
+        public String getScript() {
+            return script;
+        }
+
+        public int getPriority() {
+            return priority;
+        }
+    }
+
+    /**
+     * Script implementing the toHtml() method as an external script resource.
+     */
+    static public class FileScript extends Script {
+
+        public FileScript(String script, int priority) {
+            super(script, priority);
+        }
+
+        @Override
+        public String toHtml() {
+            StringBuilder builder = new StringBuilder();
+            builder.append("\t<script type=\"text/javascript\" src=\"");
+            builder.append(this.script);
+            builder.append("\" data-priority=\"");
+            builder.append(this.priority);
+            builder.append("\"></script>\n");
+            return builder.toString();
+        }
+
+    }
+
+    /**
+     * Script implementing the toHtml() method as an inline script resource.
+     */
+    static public class InlineScript extends Script {
+
+        public InlineScript(String script, int priority) {
+            super(script, priority);
+        }
+
+        @Override
+        public String toHtml() {
+            StringBuilder builder = new StringBuilder();
+            builder.append("\t<script type=\"text/javascript\" data-priority=\"");
+            builder.append(this.priority);
+            builder.append("\">/* <![CDATA[ */");
+            builder.append(this.script);
+            builder.append("\n/* ]]> */</script>\n");
+            return builder.toString();
+        }
+    }
+
+    protected static final Map<String, Script> SCRIPTS = new TreeMap<>();
+
+    /**
+     * Aliases for the page scripts. The path in the FileScript is relatively to
+     * the request's context path.
+     *
+     * @see HttpServletRequest#getContextPath()
+     */
+    static {
+        SCRIPTS.put("jquery", new FileScript("js/jquery-3.2.0.min.js", 10));
+        SCRIPTS.put("jquery-ui", new FileScript("js/jquery-ui-1.12.0-custom.min.js", 11));
+        SCRIPTS.put("jquery-tablesorter", new FileScript("js/jquery-tablesorter-2.26.6.min.js", 12));
+        SCRIPTS.put("tablesorter-parsers", new FileScript("js/tablesorter-parsers-0.0.1.js", 13));
+        SCRIPTS.put("searchable-option-list", new FileScript("js/searchable-option-list-2.0.3.min.js", 14));
+        SCRIPTS.put("utils", new FileScript("js/utils-0.0.9.js", 15));
+        SCRIPTS.put("repos", new FileScript("js/repos-0.0.1.js", 20));
+        SCRIPTS.put("diff", new FileScript("js/diff-0.0.1.js", 20));
+    }
+
+    /**
+     * Scripts which will be written to the page.
+     */
+    private final List<Script> outputScripts = new ArrayList<>();
+
+    /**
+     * Sort the page scripts according to the priority.
+     */
+    public void sort() {
+        outputScripts.sort((a, b) -> a.getPriority() - b.getPriority());
```

Comment:
changed

Scripts are added into the array in a sorted manner using `binarySearch` to find the insertion point

This is not the most efficient way (`O(n)`) however most readable (than using `Map<int, List<Script>>`) and for small number of scripts (which is expected) this does not really matter